core/src/api/api.rs

```diff
@@ -23,6 +24,7 @@ pub struct GraphState<E: EncryptionBehavior, const MAX_USERS: usize = MAX_GRAPH_
 	phantom: PhantomData<E>,
 	environment: Environment,
 	public_key_manager: Rc<RefCell<PublicKeyManager>>,
+	pri_manager: Rc<RefCell<PriManager>>,
```

Comment:
This has me thinking; should we adopt this approach for the Environment that gets passed down into all of the graphs? Thinking that could be a bit heavyweight due to holding multiple complete Avro schemas in each copy. Maybe something to file a task for later.

Reply:
Since we only used the reference to config in the environment. The overhead is minuscule.